### PR TITLE
pip: fix issues with 3.6 being dropped

### PR DESCRIPTION
**Pull request by henryiii**
Requested in https://github.com/conda-forge/pip-feedstock/pull/88

### CODE REVIEW

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -653,12 +653,17 @@ def _gen_new_index(repodata, subdir):
                 i = record['depends'].index('python')
                 record['depends'][i] = 'python >=3.5'
 
-        # pip >=21 requires python >=3.6 but the first build has >=3
-        # https://github.com/conda-forge/pip-feedstock/pull/68
         if record_name == 'pip':
-            if record['version'] == "21.0" and record['build'] == "pyhd8ed1ab_0":
+            # pip >=21 requires python >=3.6 but the first build has >=3
+            # https://github.com/conda-forge/pip-feedstock/pull/68
+            if record['version'] == "21.0" and ecord['build'] == "pyhd8ed1ab_0"
```
Comment (jakirkham) [inline recipe/gen_patch_json.py:659]:
```suggestion
            if record['version'] == "21.0" and record['build'] == "pyhd8ed1ab_0"
```

File: recipe/gen_patch_json.py [hunk 2]
```diff
@@ -653,12 +653,17 @@ def _gen_new_index(repodata, subdir):
                 i = record['depends'].index('python')
                 record['depends'][i] = 'python >=3.5'
 
-        # pip >=21 requires python >=3.6 but the first build has >=3
-        # https://github.com/conda-forge/pip-feedstock/pull/68
         if record_name == 'pip':
-            if record['version'] == "21.0" and record['build'] == "pyhd8ed1ab_0":
+            # pip >=21 requires python >=3.6 but the first build has >=3
+            # https://github.com/conda-forge/pip-feedstock/pull/68
+            if record['version'] == "21.0" and record['build'] == "pyhd8ed1ab_0"
```
Comment (henryiii) [inline recipe/gen_patch_json.py:659]:
```suggestion
            if record['version'] == "21.0" and record['build'] == "pyhd8ed1ab_0":
```

File: recipe/gen_patch_json.py [hunk 3]
```diff
@@ -653,12 +653,17 @@ def _gen_new_index(repodata, subdir):
...
-        # https://github.com/conda-forge/pip-feedstock/pull/68
         if record_name == 'pip':
+            # pip >=21 requires python >=3.6 but the first build has >=3
+            # https://github.com/conda-forge/pip-feedstock/pull/68
             if record['version'] == "21.0" and record['build'] == "pyhd8ed1ab_0":
                 i = record['depends'].index('python >=3')
                 record['depends'][i] = 'python >=3.6'
+            # Same issue with pip 22 moving to Python 3.7+
+            # https://github.com/conda-forge/pip-feedstock/pull/88
+            if record['version'] in {"22.0", "22.0.1", "22.0.2"} and record['build'] in {"pyhd8ed1ab_0", "pyhd8ed1ab_0", "pyhd8ed1ab_0"}:
```
Comment (henryiii) [inline recipe/gen_patch_json.py:664]:
```suggestion
            if record['version'] in {"22.0", "22.0.1", "22.0.2"} and record['build'] == "pyhd8ed1ab_0"}:
```

Just realized these are all the same.

**Comment by jakirkham** [comment]
Thoughts @conda-forge/core?

**Comment by chrisburr** [comment]
@henryiii Can you post the ouput of running `recipe/show_diff.py` with this PR?

**Comment by beckermr** [comment]
If you merge master into this branch, the show diff script will run in the CI now.

**Comment by chrisburr** [comment]
@beckermr I'm failing to find where that is shown. Can you point me to what's changed? The closest I can see is:

https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/16d415321c16aa8f953fd36250819110f9450540/recipe/meta.yaml#L43

**Comment by beckermr** [comment]
Woops! I just merged the PR with it. rebase onto master and you'll get the changes

**Pull request merged**